src/test/ui/local-ident.stderr

```diff
@@ -0,0 +1,24 @@
+error[E0308]: mismatched types
+  --> $DIR/local-ident.rs:15:16
+   |
+15 |     let x: X = S;
+   |                ^ expected struct `Mod2::X`, found struct `Mod1::S`
+   |
+   = note: expected type `X`
+              found type `S`
+
+error[E0308]: mismatched types
+  --> $DIR/local-ident.rs:16:28
+   |
+16 |     let y: Option<usize> = Ok(2);
+   |                            ^^^^^ expected enum `std::option::Option`, found enum `std::result::Result`
+   |
+   = note: expected type `Option<usize>`
+              found type `Result<{integer}, _>`
```

Comment:
Should we add a `note` here stating `Option: std::option::Option` and `Result: std::result::Result`? This could become a very long list for cases like the one put forward in #40186.